internal/state/graph/httproute.go

```diff
@@ -405,20 +406,31 @@ func findAcceptedHostnames(listenerHostname *v1beta1.Hostname, routeHostnames []
 		if hostname == "" {
 			return true
 		}
-		return string(h) == hostname
+
+		routeHost := string(h)
+		return routeHost == hostname || wildcardMatch(hostname, routeHost) || wildcardMatch(routeHost, hostname)
 	}
 
 	var result []string
 
 	for _, h := range routeHostnames {
 		if match(h) {
-			result = append(result, string(h))
+			if len(hostname) > len(h) {
```

Comment:
Are we checking the length to determine which hostname is more specific? If so, a function might help with readability.